### ISSUE TITLE
dynamic import(...).then(...) not working with swcMinify: true

### ISSUE THREAD

**Issue by eoneill** [vercel/next.js#33025]
### Run `next info` (available from version 12.0.8 and up)

```
    Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64
    Binaries:
      Node: 14.15.4
      npm: 6.14.10
      Yarn: 1.22.11
      pnpm: N/A
    Relevant packages:
      next: 12.0.8-canary.17  // actual: 12.0.7
      react: N/A              // actual: 17.0.2
      react-dom: N/A          // actual: 17.0.2
```

### What version of Next.js are you using?

12.0.7

### What version of Node.js are you using?

14.15.4

### What browser are you using?

Edge / Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Netlify

### Describe the Bug

Using dynamic `import(...).then(...)` (with promise chaining) does not work with `swcMinify: true`, but works as expected with Terser.

Additionally, using `await import(...)` with `swcMinify: true` seems to work fine.

### Expected Behavior

`import(...).then(...)` should work with `swcMinify: true`

### To Reproduce

with `swcMinify: true`...

this works:
```js
import dynamic from 'next/dynamic';
const NetlifyCMS = dynamic(
  async function lazyCMS() {
    const cms = await import('netlify-cms-app');
    cms.init({});
  },
  { ssr: false }
);

export default function AdminPage() {
  return (
    <>
      <NetlifyCMS />
    </>
  );
}
```

but this does not:
```js
const NetlifyCMS = dynamic(
  function lazyCMS() {
    return import('netlify-cms-app').then(cms => {
      cms.init({})
    });
  },
  { ssr: false }
);
```

It does not produce any build or runtime errors as far as I could tell.

Setting `swcMinify: false`, both versions work.

**Comment by balazsorban44** [comment]
Side question, when you ran `next info` and paste the result, what did you mean by the `// actual` comments?

**Comment by balazsorban44** [comment]
So I tried to reproduce this, but I'm not sure I was able to. Here is what I tried:

```js
// pages/index.js

import dynamic from "next/dynamic"

const DynamicImport = dynamic(
  function lazyCMS() {
    return import("../test").then((cms) => {
      cms.init({})
      return cms.default
    })
  },
  { ssr: false }
)

export default function Page() {
  return <DynamicImport />
}

```

```js
//test.js

export default function Test() {
  return <div>Hello World</div>
}
export function init() {
  console.log("initialized")
}
```

And I added `swcMinify: true` to `next.config.js`.

After building with `next build` and run with `next start`, I did see the Hello world component appear, and `"initialized"` being printed on the console.

So what's the actual issue? Could you add a more elaborate reproduction?

**Comment by eoneill** [comment]
Thanks for the reply and taking the time to try attempt a reproduction. My apologies for not providing thorough enough details.

I think the root issue was that with `swcMinify: true`, the module was being (correctly) dead-code eliminated. FWIW, it worked in dev, and failed in prod.

Anyway, I ended up getting it all working and am actually using `dynamic()` to render a proper dynamic component now. So marking this as closed.

Thanks again for the friendly response and your time.



> Side question, when you ran `next info` and paste the result, what did you mean by the `// actual` comments?

I added the "actual" versions I had because `next info` didn't detect them (returned `N/A`).
